lib/CMakeLists.txt

```diff
@@ -166,6 +166,68 @@ target_link_libraries(hermesvmlean_a PUBLIC
 # CMake requires at least one file for the target, so create an empty file.
 add_library(hermesvm SHARED dummy.cpp)
 add_library(hermesvmlean SHARED dummy.cpp)
+
+# Do not build a framework for the test suite.
+if(NOT HERMES_ENABLE_TEST_SUITE)
+  if(APPLE)
```

Comment:
Yes, it seems like can solve problem 👍

Reply:
I'll check it 👍

Reply:
It will work if I change the default value, as shown in this commit: https://github.com/facebook/hermes/pull/1566/commits/c384c05a5c4aeb51daf7f00654276c5d4b4db8c0

Reply:
Why does the default need to change? Is there a specific error you get?

I'm curious, since in the original Hermes build, the check seems to work fine.

Reply:
Without that change, Hermes's unit tests fail with an error:
```
--
Exit Code: 2

Command Output (stderr):
--
ld: library not found for -lhermesvm
clang: error: linker command failed with exit code 1 (use -v to see invocation)
/usr/bin/cc: execution failed
FileCheck error: '-' is empty.
FileCheck command line:  /Users/distiller/project/build/bin/./FileCheck --match-full-lines /Users/distiller/project/hermes/test/shermes/regress-relational-simplify.js
```
more logs: https://app.circleci.com/pipelines/github/facebook/hermes/7164/workflows/ef45f59e-489a-449b-8d75-4bc2501a27db/jobs/72562

It seems like a unit test misconfiguration, but of course, if you want, I can investigate it further of course

Reply:
Ah, interesting, it's the `shermes` tests that start failing. That makes sense, I agree we should change the default to OFF.